### PR TITLE
Clean-up Python #4815 - P1.14: Chem\Scaffolds & Chem\SimpleEnum

### PR DESCRIPTION
**Pull request by IchiruTake**
This is the sub-PR breaking from the large PR 4815: https://github.com/rdkit/rdkit/pull/4815

-----------------------
[Remark]: These two modules are completely independence. We only merged it due to the simplicity and easy clean-up. No intention here

### CODE REVIEW

File: rdkit/Chem/Scaffolds/MurckoScaffold.py [hunk 1]
```diff
@@ -75,6 +75,7 @@ def GetScaffoldForMol(mol):
     res.UpdatePropertyCache()
     Chem.GetSymmSSSR(res)
   else:
+    # This code cannot be reached
     res = _pyGetScaffoldForMol(mol)
```
Comment (gedeck) [inline rdkit/Chem/Scaffolds/MurckoScaffold.py:79]: The original Python implementation was at some point replaced by a C++ implementation. @greglandrum, what do you think about removing the old code?

File: rdkit/Chem/SimpleEnum/Enumerator.py [hunk 2]
```diff
@@ -214,21 +216,17 @@ def _uniqueOnly(lst):
...
           seen.append(smi)
           yield entry
-
-  ps = AllChem.EnumerateLibraryFromReaction(reaction, bbLists)
-  if not uniqueProductsOnly:
-    return ps
-  else:
-    return _uniqueOnly(ps)
+  
+  return _uniqueOnly(ps)
```
Comment (gedeck) [inline rdkit/Chem/SimpleEnum/Enumerator.py:220]: Not sure if this change adds anything.

**Review by gedeck — CHANGES_REQUESTED** [review]

File: rdkit/Chem/SimpleEnum/Enumerator.py (hunk 2 above)
Comment (IchiruTake) [reply rdkit/Chem/SimpleEnum/Enumerator.py:220]: The old required the function `_uniqueOnly()` must be called regardless of consequences. This ensure we don't mess thing with. I will upgrade it

**Review by gedeck — APPROVED** [review]

File: rdkit/Chem/Scaffolds/MurckoScaffold.py (hunk 1 above)
Comment (greglandrum) [reply rdkit/Chem/Scaffolds/MurckoScaffold.py:79]: In general we've left the python implementations in place. I suppose there's not really an argument against removing them at some point. That's a different PR though. :-)

**Review by greglandrum — APPROVED** [review]
LGTM

**Pull request merged**